developerportal/apps/ingestion/utils.py

```diff
@@ -50,6 +52,41 @@ def _get_item_image(entry) -> str:
     return ""
 
 
+def _get_item_description(entry) -> str:
+    """If an entry has a description field, grab it, suitable for use in
+    a RichText field"""
+
+    desc = ""
+
+    if entry and getattr(entry, "description") is not None:
+
+        # If it's a multi-line description, we only want the first
+        desc = entry.description.split("\n")[0]
+
+        # Extract the string within the first <p>, if any, in the content
+        # (eg this is how Wordpress appears to format it for Mozilla Hacks)
+        soup = BeautifulSoup(desc, features="lxml")
+        try:
+            desc = soup.find("p").text
+        except AttributeError as ae:
```

Comment:
I'm of the "don't-use-error-handling-for-that-which-can-be-sanely-checked" school, so I'd personally go with something like:
```python
if text in paragraph:
  desc = paragraph.text
else:
  desc = None
  logger.warning("yadayada")
```

But yours might be more pythonic, I don't know much about pythonicisim

Reply:
Yep I hear you, and thanks! I went that way because it would be pretty unusual not to get that <p> node, but I realise I do prefer sit on the exceptions-as-control-flow side of the aisle in the Python church, partly because there's (a tiny bit) less code. I think I'll stick with it for now, but I do get your point.